glitch/parsers/cmof.py

```diff
@@ -97,56 +97,92 @@ def __get_element_code(start_token, end_token, code):
 
     @staticmethod
     def __parse_vars(unit_block, cur_name, token, code):
-        def create_variable(name, value):
-            has_variable = ("{{" in value) and ("}}" in value)
+        def create_variable(token, name, value, child=False) -> Variable:
+            has_variable = (("{{" in value) and ("}}" in value)) if value != None else False
             if (value in ["null", "~"]): value = ""
             v = Variable(name, value, has_variable)
             v.line = token.start_mark.line + 1
-            v.code = AnsibleParser.__get_element_code(token, value, code)
+            if value == None:
+                v.code = AnsibleParser.__get_element_code(token, token, code)
+            else:
+                v.code = AnsibleParser.__get_element_code(token, value, code)
             v.code = ''.join(code[token.start_mark.line : token.end_mark.line + 1])
-            unit_block.add_variable(v)
 
+            variables.append(v)
+            if not child:
+                unit_block.add_variable(v)
+            return v
+
+        def process_var(key, v, child: bool = False):
+            if isinstance(v,  MappingNode):
+                var = create_variable(v, key.value, None, child)
+                aux_vars = []
+                aux_vars += AnsibleParser.__parse_vars(
+                    unit_block, key.value, v, code)
+                var.variables = aux_vars
```

Comment:
Why not `var.variables.append(AnsibleParser.__parse_vars(unit_block, key.value, v, code))` ?

In any case, if you prefer to use the auxiliary variable, you can do `aux_vars = AnsibleParser...` (without the assignment at line 119)

Reply:
I Agree, Thank you!